ecs/pod_creator.go

```diff
@@ -27,11 +31,227 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
 
 // CreatePod creates a new pod backed by AWS ECS.
 func (m *BasicECSPodCreator) CreatePod(ctx context.Context, opts ...*cocoa.ECSPodCreationOptions) (cocoa.ECSPod, error) {
-	return nil, errors.New("TODO: implement")
+
+	mergedPodCreationOpts := cocoa.MergeECSPodCreationOptions(opts...)
+	mergedPodExecutionOpts := cocoa.MergeECSPodExecutionOptions(mergedPodCreationOpts.ExecutionOpts)
+
+	err := mergedPodCreationOpts.Validate()
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid pod creation options")
+	}
+
+	err = mergedPodExecutionOpts.Validate()
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid pod execution options")
+	}
+
+	allEnvVars, allSecrets, allSecretsToCreate, err := m.exportSecrets(ctx, mergedPodCreationOpts)
+	if err != nil {
+		return nil, errors.Wrap(err, "exporting secrets")
+	}
+
+	err = m.createSecrets(ctx, allSecretsToCreate)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating secrets")
+	}
+
+	taskDefinition, err := m.exportPodCreationOptions(ctx, mergedPodCreationOpts, allEnvVars, allSecrets)
+	if err != nil {
+		return nil, errors.Wrap(err, "translating the register task definition input to the correct type")
+	}
+
+	registerOut, err := m.client.RegisterTaskDefinition(ctx, taskDefinition)
+	if err != nil {
+		return nil, errors.Wrap(err, "registering task definition")
+	}
+
+	if registerOut.TaskDefinition == nil || registerOut.TaskDefinition.TaskDefinitionArn == nil {
+		return nil, errors.New("missing task definition")
```

Comment:
It might be more useful to the end user to have more details than this - if they just see this message, they won't know what actually went wrong. In these cases, you should say something to the effect of "we can't continue because something violated our expectations". I would phrase this error as something like "expected a task definition from ECS, but none was returned".